### PR TITLE
Ea 2394 exposing metrics

### PR DESCRIPTION
**Pull request by SrishtiSingh-eu**

### CODE REVIEW

**Review by gsergiu — CHANGES_REQUESTED** [review]
I left comments related to some small (non functional) issues behind , but there is also an important functional issue related to the computation for the number of liked items metric, which needs to be addressed.

File: set-usage-stats/pom.xml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    <name>Europeana (User) Set - Usage Stats</name>
+    <organization>
+        <name>Europeana Foundation + AIT</name>
+    </organization>
+    <url>http://www.europeanalabs.eu</url>
+
+    <dependencies>
+        <!-- europeana api commons -->
+
+        <dependency>
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-usage-stats/pom.xml:23]: done

File: set-usage-stats/pom.xml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            <groupId>eu.europeana.api.commons</groupId>
+            <artifactId>commons-definitions</artifactId>
+            <version>${version.commonsApi}</version>
+        </dependency>
+
+        <!-- set projects -->
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-usage-stats/pom.xml:29]: done

File: set-usage-stats/src/main/java/eu/europeana/set/stats/service/UsageStatsService.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+import org.apache.logging.log4j.Logger;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.Resource;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+
+@Service
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-usage-stats/src/main/java/eu/europeana/set/stats/service/UsageStatsService.java:22]: done. have configured it as a bean in set-web xml file.

File: set-usage-stats/src/main/java/eu/europeana/set/stats/service/UsageStatsService.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+public class UsageStatsService {
+
+    @Resource
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-usage-stats/src/main/java/eu/europeana/set/stats/service/UsageStatsService.java:25]: done

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/UsageStatsUserSetRest.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+@Controller
+@SwaggerSelect
+@Api(tags = "User Set Usage Statistics API")
+@ComponentScan("eu.europeana.set.stats")
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/UsageStatsUserSetRest.java:30]: done

File: set-usage-stats/src/main/java/eu/europeana/set/stats/service/UsageStatsService.java [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+          averageUserSetsPerUser = totalUserSets / distinctUsers;
+      }
+      metric.setAverageSetsPerUser(averageUserSetsPerUser);
+    }
+
+    /**
+     * Returns the current date time in ISO format
+     * @return
+     */
+    public String getCurrentISODate() {
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-usage-stats/src/main/java/eu/europeana/set/stats/service/UsageStatsService.java:94]: have removed it

File: set-usage-stats/src/main/java/eu/europeana/set/stats/vocabulary/UsageStatsFields.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    public static final String OVERALL_TOTAL_TYPE = "OverallTotal";
+    public static final String TYPE               = "type";
+    public static final String CREATED            = "created";
+
+    public static final String NUMBER_OF_PRIVATE_SETS = "NumberOfPrivateSets";
+    public static final String NUMBER_OF_PUBLIC_SETS = "NumberOfPublicSets";
+    public static final String NUMBER_OF_ITEMS_LIKED = "NumberOfItemsLiked";
+    public static final String AVERAGE_SETS_PER_USER = "AverageSetsPerUser";
+
+    public static final String DATE_FORMAT = "yyyy-MM-dd'T'HH:mm'Z'";
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-usage-stats/src/main/java/eu/europeana/set/stats/vocabulary/UsageStatsFields.java:17]: removed

File: set-usage-stats/src/main/java/eu/europeana/set/stats/vocabulary/UsageStatsFields.java [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+public class UsageStatsFields {
+
+    UsageStatsFields() {
+    }
+
+    public static final String OVERALL_TOTAL_TYPE = "OverallTotal";
+    public static final String TYPE               = "type";
+    public static final String CREATED            = "created";
+
+    public static final String NUMBER_OF_PRIVATE_SETS = "NumberOfPrivateSets";
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-usage-stats/src/main/java/eu/europeana/set/stats/vocabulary/UsageStatsFields.java:12]: done

File: set-usage-stats/src/main/java/eu/europeana/set/stats/service/UsageStatsService.java [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+    }
+
+    /**
+     * Gets the total item liked which is the total number
+     * of item in all BookmarkFolders
+     *
+     * @param metric
+     */
+    public void getTotalItemsLiked(Metric metric) {
+      long totalItemsLiked = 0;
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-usage-stats/src/main/java/eu/europeana/set/stats/service/UsageStatsService.java:60]: done

**Pull request merged**